### PR TITLE
fixes about 503 blocks, removed deprecated libs, general updates

### PR DESCRIPTION
**Pull request by OlehVeselov92**
- fixed issue with 503 blocks instead of normal response with JSON info;
- Updated SDK to 1.2.0 (+ General fixes regarding new version);
- removed usage of deprecated "request-promise" library (using Apify.utils.requestAsBrowser() instead);
- implemented usage of proxy (auto configurations) by default;

### CODE REVIEW

File: apify_storage/key_value_stores/default/OUTPUT.json [hunk 1]
```diff
@@ -11,55 +11,14 @@
       "state": "Prague",
```
Comment (pocesar) [inline apify_storage/key_value_stores/default/OUTPUT.json]: delete the file after setting the folder to gitignore

File: main.js [hunk 2]
```diff
@@ -19,49 +18,27 @@ Apify.main(async () => {
...
                 url: `https://www.kickstarter.com/locations/search?searchable=true&term=${query}`,
+                proxyUrl: proxyConfiguration.newUrl(),
                 headers: {
-                    accept: 'application/json, text/javascript, */*; q=0.01',
-                    'accept-language': 'en;q=0.9',
-                    referer: url,
-                    'x-csrf-token': csrfToken,
-                    'x-requested-with': 'XMLHttpRequest',
+                    // eslint-disable-next-line max-len
+                    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36',
```
Comment (pocesar) [inline main.js]: I think that using got-scraping internally by requestAsBrowser, passing the headers manually isn't the best way. only `Accept: 'application/json'` should remain

File: main.js [hunk 3]
```diff
@@ -19,49 +18,27 @@ Apify.main(async () => {
...
     const { query } = input;
     if (!query || typeof query !== 'string') crash('Query loaded from INPUT is empty or not a string.');
-
-    const commonHeaders = {
-        'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36',
-    };
-
     let attempts = 20;
     let response;
     do {
```
Comment (pocesar) [inline main.js]: this loop could become a CheerioCrawler / BasicCrawler instance, the code is very much optimistic that things will just work (no blocks, proper json payload received, 200 http code, etc)

**Review by pocesar — CHANGES_REQUESTED** [review]

**Comment by metalwarrior665** [comment]
Wow, "apify": "^0.9.15", that is the record :D

File: main.js [hunk 4]
```diff
@@ -1,89 +1,54 @@
...
-                    'x-requested-with': 'XMLHttpRequest',
+                    accept: 'application/json',
                 },
-                json: true,
             });
-
-            if (!response.locations) crash('Response does not contain locations. Try again.');
-
+            const response = JSON.parse(body);
+            if (!response.locations) log.error('Response does not contain locations. Try again.');
```
Comment (pocesar) [inline main.js]: throw here for retrying

File: main.js [hunk 5]
```diff
@@ -1,89 +1,54 @@
...
+    await requestQueue.addRequest({
+        url,
+    });
+    log.info(`${url} was added to the queue`);
+    const crawler = new Apify.BasicCrawler({
+        requestQueue,
+        handleRequestFunction: async ({ request }) => {
+            const { body } = await Apify.utils.requestAsBrowser({
+                url: request.url,
+                proxyUrl: proxyConfiguration.newUrl(),
```
Comment (pocesar) [inline main.js]: `proxyConfiguration.newUrl(session.id)`

File: main.js [hunk 6]
```diff
@@ -1,89 +1,54 @@
...
-}
+const { log } = Apify.utils;
 
 Apify.main(async () => {
-    // Load query from input
+    const proxyConfiguration = await Apify.createProxyConfiguration();
+    const requestQueue = await Apify.openRequestQueue();
     const input = await Apify.getValue('INPUT');
-    if (!input) crash('Key-value store does not contain INPUT.');
+    if (!input) log.error('Can not get input');
```
Comment (pocesar) [inline main.js]: throw instead of log

File: main.js [hunk 7]
```diff
@@ -1,89 +1,54 @@
...
-
-            const $ = cheerio.load(html);
-            const csrfToken = $('meta[name="csrf-token"]').attr('content');
-
-            if (!csrfToken) crash('Could not load CSRF token. Try again.');
-
-            // Query the kickstarter location search page with the correct csrf token and cookies
-            response = await preparedRequest({
-                url: `https://www.kickstarter.com/locations/search?searchable=true&term=${query}`,
+    if (!query || typeof query !== 'string') log.error('Query loaded from INPUT is empty or not a string.');
```
Comment (pocesar) [inline main.js]: same

**Review by pocesar — CHANGES_REQUESTED** [review]
thanks man! almost there

**Review by pocesar — APPROVED** [review]

**Comment by pocesar** [comment]
I'm going to run here in a bit to see how is everything and will release in a bit

**Pull request merged**